.github/workflows/build-binary.yaml

```diff
@@ -0,0 +1,216 @@
+name: Build binaries
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        edition: [ "", "-pm" ]
+        target:
+          - x86_64-unknown-linux-gnu
+          - aarch64-unknown-linux-gnu
+          - x86_64-unknown-linux-musl
+          - aarch64-unknown-linux-musl
+          - x86_64-apple-darwin
+          - aarch64-apple-darwin
+          - x86_64-pc-windows-msvc
+
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-22.04
+            install: |
+              sudo apt install -y libssl-dev
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-22.04
+            cross: "true"
+            # Cross' Ubuntu container is based on 20.04. Its OpenSSL version is too old for us.
+            args: --features vendored
+
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-22.04
+            args: --features vendored
+            install: |
+              sudo apt install -y musl-tools
+          - target: aarch64-unknown-linux-musl
+            os: ubuntu-22.04
+            cross: "true"
+            args: --features vendored
+
+          - target: x86_64-apple-darwin
+            os: macos-12
+          - target: aarch64-apple-darwin
+            os: macos-14
+
+          - target: x86_64-pc-windows-msvc
+            os: windows-2022
+            ext: ".exe"
+            archive: zip
+            install: |
+              git config --system core.longpaths true
+              echo "VCPKG_ROOT=$env:VCPKG_INSTALLATION_ROOT" | Out-File -FilePath $env:GITHUB_ENV -Append
+              vcpkg install openssl:x64-windows-static-md
+
+    env:
+      # name of the binary output by the build
+      output_binary_name: "trustd"
+      # name of the binary available for download
+      download_binary_name: "trustd${{ matrix.edition }}"
+      dirname: "trustd${{ matrix.edition }}-${{ inputs.version }}-${{ matrix.target }}"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup | Cache Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: "${{ matrix.target }}-release${{ matrix.edition }}-${{ hashFiles('**/Cargo.lock') }}"
```

Comment:
```suggestion
        uses: Swatinem/rust-cache@v2
```

Reply:
Doesn't that conflict with the other cache? Let me read into the use of that.

Reply:
Looks like that requires the use of the `shared-key` argument, which I added.